scene/3d/node_3d.cpp

```diff
@@ -845,6 +846,84 @@ void Node3D::_validate_property(PropertyInfo &property) const {
 	}
 }
 
+bool Node3D::property_can_revert(const String &p_name) {
+	if (p_name == "basis") {
+		return true;
+	} else if (p_name == "scale") {
+		return true;
+	} else if (p_name == "quaternion") {
+		return true;
+	} else if (p_name == "rotation") {
+		return true;
+	} else if (p_name == "position") {
+		return true;
+	}
+	return false;
+}
+
+Variant Node3D::property_get_revert(const String &p_name) {
+	Variant r_ret;
+	bool valid = false;
+
+	if (p_name == "basis") {
+		Variant variant = PropertyUtils::get_property_default_value(this, "transform", &valid);
+		if (valid) {
+			if (variant.get_type() == Variant::Type::TRANSFORM3D) {
```

Comment:
Suggestion: make these checks more compact by writing them as `if (valid && variant.get_type() == Variant::Type::...)` so that only one `else` branch is needed.